src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -1438,29 +1438,50 @@ void GameObject::ActivateObject(GameObjectActions action, WorldObject* spellCast
 
     switch (action)
     {
+        case GameObjectActions::None:
+            TC_LOG_FATAL("spell", "Spell %d has action type NONE in effect %d", spellId, effectIndex);
+            break;
         case GameObjectActions::AnimateCustom0:
         case GameObjectActions::AnimateCustom1:
         case GameObjectActions::AnimateCustom2:
         case GameObjectActions::AnimateCustom3:
             SendCustomAnim(uint32(action) - uint32(GameObjectActions::AnimateCustom0));
             break;
         case GameObjectActions::Disturb: // What's the difference with Open?
+            if (unitCaster)
+                Use(unitCaster);
+            break;
+        case GameObjectActions::Unlock:
+        case GameObjectActions::Lock:
+            ApplyModFlag(GAMEOBJECT_FLAGS, GO_FLAG_LOCKED, action == GameObjectActions::Lock);
+            break;
         case GameObjectActions::Open:
             if (unitCaster)
                 Use(unitCaster);
             break;
         case GameObjectActions::OpenAndUnlock:
             if (unitCaster)
+            {
                 UseDoorOrButton(0, false, unitCaster);
-            [[fallthrough]];
-        case GameObjectActions::Unlock:
-        case GameObjectActions::Lock:
-            ApplyModFlag(GAMEOBJECT_FLAGS, GO_FLAG_LOCKED, action == GameObjectActions::Lock);
+                RemoveFlag(GAMEOBJECT_FLAGS, GO_FLAG_LOCKED);
```

Comment:
Is RemoveFlag working better in this case than ApplyModFlag ?

Reply:
It's single case now